Fix creation/displaying of projects

TomlDev opened the pull request:
### Changed

- Floating buttons overflow
![grafik](https://github.com/user-attachments/assets/1a30fcb1-4038-4c6c-a02e-075e36fff6c9)
- Added missing fields to projects:
  - mailalias, maillist, mailquota, proxyAddresses, quota
- Separated isLoading states in class management store
- Added form error to group dialog
- Fix displayName not initialized in group dialog
- Updated group form schema to validate new fields


#### Known issues
- Update/patch of project is not possible -> AK will check

michidold commented on libs/src/groups/constants/groupFormSchema.ts:
```diff
@@ -18,7 +25,27 @@ const getGroupFormSchema = (t: TFunction<'translation', undefined>) =>
+    mailquota: z.number(),
+    maillist: z.boolean(),
+    mailalias: z.boolean(),
+    quota: z.string().refine(
```
There is a false error:
![Bildschirmfoto vom 2025-01-13 09-49-50](https://github.com/user-attachments/assets/f03d7ebe-0d07-46e4-b41a-e68c07871808)

michidold commented on apps/frontend/src/pages/ClassManagement/components/GroupDialog/GroupDialog.tsx:
I would rename the button to "Edit Session" if its saved, because you can also remove the session in this dialog:
![grafik](https://github.com/user-attachments/assets/0165bc61-4728-469c-848a-4f09ae0c90fb)

michidold reviewed (changes requested):
Some findings, see also in the Edit Project dialog, there are still some issues, like:
- When opening, the name field is not set
- for proxy address there is no validation
- the mail quota expects a number but interprets it as string
- If there is a proxyAddress set, its not safed or just not filled by reopening

michidold commented:
Can you also fix these inconsistancies:
Change to our button component, with small size:
![grafik](https://github.com/user-attachments/assets/45378e7a-2bde-49b2-b7a7-e69e6818fd7c)
Change to small size:
![grafik](https://github.com/user-attachments/assets/33295e7d-3743-413d-8770-035a97b1d736)

TomlDev commented on apps/frontend/src/locales/de/translation.json:
```diff
@@ -323,7 +323,12 @@
+    "invalidQuota": "Die Quota muss der Dokumentation entsprechend angegeben werden",
     "name": "Name",
+    "mailQuota": "Mail Quota",
+    "proxyAddresses": "Proxy Adressen",
```
Renamed to Proxy Email-Adressen

TomlDev replied:
Added a placeholder example
![grafik](https://github.com/user-attachments/assets/e576a656-92f6-45cb-8ee2-7cf17cc26950)
And for proxy addresses this
![grafik](https://github.com/user-attachments/assets/430b3448-a2b3-45cb-98d2-285588cd58f4)

For quota its too complex, this is the description from LMN:

> Attributes for quota
> 
>     mailquota is an attribute to add quota for email, as integer (e.g. "mailquota": 100)
>     quota is an attribute to add quota per share, as an object. Examples:
> 
> "quota": [{"quota":50,"share":"default-school"]
> 
> or
> 
> "quota": [{"quota":500,"share":"agy"}, {"quota":300,"share":"linuxmuster-global"}]


I have created a new task for this https://github.com/edulution-io/edulution-ui/issues/370

TomlDev commented on apps/frontend/src/locales/en/translation.json:
```diff
@@ -323,6 +323,11 @@
     "title": "CLASS ROOM",
     "sidebar": "Class Room",
+    "invalidProxyAddresses": "Proxy-addresses must be written comma separated without spaces",
+    "invalidQuota": "Die Quota muss der Dokumentation entsprechend angegeben werden",
+    "mailQuota": "Mail Quota",
+    "proxyAddresses": "Proxy Adressen",
```
Changed to Proxy Email-Adresses

michidold replied:
Still german text here.

TomlDev commented:
> Some findings, see also in the Edit Project dialog, there are still some issues, like:
> 
>     * When opening the name field is not set
There seems to be a new bug introduced by changes in the LMN API, I reported it.
>     * for proxy address there is no validation
Added 👍🏽 
>     * the mail quota expects a number but interprets it as string
Fixed 👍🏽 
>     * If there is a proxyAddress set, its not safed or just not filled at render
There seems to be a new bug introduced by changes in the LMN API, I reported it.

Now there is a placeholder to show an example fill out:
![grafik](https://github.com/user-attachments/assets/9a223ec4-2b7f-4dcd-95e4-13fe159b14db)

TomlDev commented on apps/frontend/src/pages/ClassManagement/LessonPage/UserArea/UserArea.tsx:
```diff
@@ -57,7 +57,7 @@ const UserArea = () => {
   const pageBarsHeight =
     useElementHeight([FLOATING_BUTTONS_BAR_ID, LESSON_SESSION_HEADER_ID, FILTER_BAR_ID, FOOTER_ID], isMemberSelected) +
-    10;
+    30;
```
I think it's not worth the effort, we must rework the design.

michidold replied:
Ok, can you add a concrete task to do so?

TomlDev replied:
I think we already had one in Jira...
New one here: https://github.com/edulution-io/edulution-ui/issues/385

michidold commented on apps/frontend/src/pages/ClassManagement/components/GroupDialog/GroupDialog.tsx:
I had to issues with this dialog. First you cant close it
[Bildschirmaufzeichnung vom 2025-01-23 10-31-05.webm](https://github.com/user-attachments/assets/2bdf499c-f61d-4e00-8ec3-52408c9241c7)
Second, on delete there is no redirect.
[Bildschirmaufzeichnung vom 2025-01-23 10-33-10.webm](https://github.com/user-attachments/assets/24b023fc-9ef5-4d63-b84f-84418b5be60f)

michidold reviewed (changes requested):
Found some stuff, please have a look

michidold reviewed (approved):
Looks good
